scripts/lib/integration.sh

```diff
@@ -29,4 +29,74 @@ function run_kops_conformance() {
     DOWN_KOPS_DURATION=$((SECONDS - START))
     echo "TIMELINE: Down KOPS cluster took $DOWN_KOPS_DURATION seconds."
     exit 0
-}
\ No newline at end of file
+}
+
+function setup_warm_ip_test() {
+    echo "Setting WARM_IP_TARGET=2 and MINUMUM_IP_TARGET=10"
+    echo "Running warm ip test"
+    $KUBECTL_PATH set env ds aws-node -n kube-system WARM_IP_TARGET=2
+    $KUBECTL_PATH set env ds aws-node -n kube-system MINIMUM_IP_TARGET=10
+    KUBECTL_WARM_IP_TARGET=$($KUBECTL_PATH describe ds -n kube-system | grep WARM_IP_TARGET)
+    KUBECTL_MINIMUM_IP_TARGET=$($KUBECTL_PATH describe ds -n kube-system | grep MINIMUM_IP_TARGET)
+    if [[ $KUBECTL_WARM_IP_TARGET != *"2"* || $KUBECTL_MINIMUM_IP_TARGET != *"10"* ]]; then
+        echo "WARM_IP_TARGET not propogated in daemonset!"
+        on_error
+    fi
+
+    FIRST_DS_POD_NAME=$($KUBECTL_PATH get pods -n kube-system | grep aws-node | sed -n '1 p' | awk '{print $1}')
+    SECOND_DS_POD_NAME=$($KUBECTL_PATH get pods -n kube-system | grep aws-node | sed -n '2 p' | awk '{print $1}')
+    WARM_IP_VALUE1=""
+    WARM_IP_VALUE2=""
+    MINIMUM_IP_VALUE1=""
+    MINIMUM_IP_VALUE2=""
+    START=$SECONDS
+    while [[ $($KUBECTL_PATH describe pod $FIRST_DS_POD_NAME -n=kube-system | grep WARM_IP_TARGET) != *"2"* || \
+        $($KUBECTL_PATH describe pod $FIRST_DS_POD_NAME -n=kube-system | grep MINIMUM_IP_TARGET) != *"10"* || \
+        $($KUBECTL_PATH describe pod $SECOND_DS_POD_NAME -n=kube-system | grep WARM_IP_TARGET) != *"2"* || \
+        $($KUBECTL_PATH describe pod $SECOND_DS_POD_NAME -n=kube-system | grep MINIMUM_IP_TARGET)  != *"10"* ]]
+    do
+        echo "Waiting for daemonset pod propagation..."
+        FIRST_DS_POD_NAME=$($KUBECTL_PATH get pods -n kube-system | grep aws-node | sed -n '1 p' | awk '{print $1}')
+        SECOND_DS_POD_NAME=$($KUBECTL_PATH get pods -n kube-system | grep aws-node | sed -n '2 p' | awk '{print $1}')
+        sleep 5
+        if [[ $((SECONDS - START)) -gt 500 ]]; then
+            echo "Propogation for WARM_IP_TARGET timed out!"
+            on_error
+        fi
+    done
+    echo "WARM_IP_TARGET and MINIMUM_IP_TARGET successfully propagated!"
+    echo $WARM_IP_VALUE1
+    echo $MINIMUM_IP_VALUE1
+}
+
+function setup_warm_eni_test() {
+    echo "Setting WARM_ENI_TARGET=0"
+    echo "Running warm eni test"
+    $KUBECTL_PATH set env ds aws-node -n kube-system WARM_ENI_TARGET=0
+    #Sleep a couple seconds to ensure propagation
+    sleep 2
+    KUBECTL_WARM_ENI_TARGET=$($KUBECTL_PATH describe ds -n kube-system | grep WARM_ENI_TARGET)
+    if [[ $KUBECTL_WARM_ENI_TARGET != *"0" ]]; then
+        echo "WARM_ENI_TARGET not propogated in daemonset!"
+        on_error
+    fi
+    FIRST_DS_POD_NAME=$($KUBECTL_PATH get pods -n kube-system | grep aws-node | sed -n '1 p' | awk '{print $1}')
+    SECOND_DS_POD_NAME=$($KUBECTL_PATH get pods -n kube-system | grep aws-node | sed -n '2 p' | awk '{print $1}')
+    WARM_ENI_VALUE1=""
+    WARM_ENI_VALUE2=""
```

Comment:
Same comment about unused WARM_ variables.

Reply:
Thanks for pointing that out, have been removed now!